Src/App/DIC.php

```diff
@@ -0,0 +1,45 @@
+<?php
+namespace Oat\App;
+
+use Exception;
+
+class DIC implements DICInterface
+{
+    private $registry = [];
+
+    /**
+     * find a closure with it class name
+     * @param $className
+     * @return mixed
+     * @throws Exception
+     */
+    public function get($shortClassName)
+    {
+        if (!array_key_exists($shortClassName, $this->registry)) {
+            throw new Exception('class ' . $shortClassName . '  isn\'t set');
+        }
+        return $this->registry[$shortClassName]();
+    }
+
+    /**
+     * stock a closure of in $this->registry with $shortClassName as key
+     * @param $className
+     * @param null $closure
+     * @throws Exception
+     */
+    public function set($className, $closure = null)
+    {
+        if (!is_callable($closure))
+        {
+            if (!class_exists($className)) {
+                throw new Exception('class ' . $className . ' doesn\'t exist');
+            }
+            $closure = function () use ($className) {
+                return new $className();
+            };
+        }
+        $modifiedClassName = array_reverse(explode('\\',$className));
```

Comment:
Can you allow to put an alias instead of classname?
You are dealing with DependencyInjection, not about ClassInjection ;p